urllib3/util/rfc6555.py

```diff
@@ -0,0 +1,223 @@
+import errno
+import socket
+from urllib3.util.selectors import (
+    DefaultSelector,
+    EVENT_WRITE
+)
+from urllib3.util.timeout import current_time
+
+
+_HAPPY_EYEBALLS_CACHE = {}
+_HAPPY_EYEBALLS_CACHE_TIME = 60 * 10  # 10 minutes according to RFC 6555
+_ASYNC_ERROR_NUMBERS = set([errno.EINPROGRESS,
+                            errno.EAGAIN,
+                            errno.EWOULDBLOCK])
+if hasattr(errno, "WSAEWOULDBLOCK"):  # Platform-specific: Windows
+    _ASYNC_ERROR_NUMBERS.add(errno.WSAEWOULDBLOCK)
+
+_IP_FAMILIES = set([socket.AF_INET])
+if hasattr(socket, "AF_INET6"):
+    _IP_FAMILIES.add(socket.AF_INET6)
```

Comment:
Here's a real question: is there any point in doing Happy Eyeballs if we don't have IPv6?